interface/main/messages/messages.php

```diff
@@ -51,19 +51,87 @@
 <head>
 
 <?php html_header_show();?>
+<style>
+  .listbox{
+    border: 2px solid #000;
+    width: 150px;
+    max-height: 125px;
+    overflow: auto;
+    background-color: #f6f6f6;
+    color: #2ECC40;
+    font-weight: bold;
+    font-size: 14px;
+  }
+  .listbox> ul{
+    list-style: none;
+    padding-left: 0;
+    margin-bottom: 0;
+  }
+  .listbox> ul> li{
+    border: 1px solid #000;
+    padding: 2px 15px;
+  }
+</style>
 </head>
 
 <body class="body_top">
 <span class="title" style="display: none"><?php echo xlt('Message and Reminder Center'); ?></span>
+<span class="title"><?php echo xlt("Who's Online?"); ?></span>
+
+<div class="listbox">
+<?php
+  //creating list for first time upon initial loading
 
+  //select users whose last activity during
+  //present date is logging in successfully
+  $query = "SELECT `user`
+            FROM `log`
+            WHERE `date` IN (SELECT  max(`date`)
+                             FROM `log`
+                             WHERE DATE_FORMAT(`date`,'%Y-%M-%d') = DATE_FORMAT(now(),'%Y-%M-%d')
+                             GROUP BY `user`)
+                  AND `event` = 'login' AND `success` = 1";
+  $query .= " LIMIT 500";
```

Comment:
Is this limit going to be an issue o the larger systems? @aethelwulffe will 500 users be enough?

Reply:
500 online users at same time seems like a reasonable limit but maybe not enough for larger staff

Reply:
500 is a crazy lot, but this can be tweaked in an in-page configuration setting.